docs/data/material/components/modal/SpringModal.tsx

```diff
@@ -7,7 +7,7 @@ import Typography from '@mui/material/Typography';
 import { useSpring, animated } from '@react-spring/web';
 
 interface FadeProps {
-  children: React.ReactElement<any>;
+  children: React.JSX.Element;
```

Comment:
I think the change of these should be

```diff
-children: React.ReactElement<any>;
+children: React.ReactElement<unknown>;
```

Why `React.JSX.Element`?

Reply:
Then we should do `React.ReactElement<{ onClick: React.MouseEventHandler}>`, because we require the children to support this prop.

Reply:
Ahhh, i had posted reason in above comment why i didn't went with `React.ReactElement<{ onClick: React.MouseEventHandler}>` somehow it didn't get saved.

Anyway, if i went with `React.ReactElement<{onClick?:React.MouseEventHandler}>` ts throwing below error here (on line 79)

https://github.com/mui/material-ui/blob/506ea4c8136117d2ef46133168d9d80bd4301cb1/docs/data/material/components/modal/SpringModal.tsx#L77-L81

`Types of property 'props' are incompatible.
                      Type 'unknown' is not assignable to type '{ onClick?: MouseEventHandler<Element> | undefined; }'.`

I wasn't sure how to fix above error hence went with `React.JSX.Element`

Reply:
Ok, at this point, I think it's better to keep `React.ReactElement<any>` in this file (which is virtually the same as having `React.JSX.Element`), so let's go with `React.ReactElement<any>` to make it explicit.